src/main/java/com/checkmarx/flow/service/SastScanner.java

```diff
@@ -71,10 +72,26 @@ public ScanResults scan(ScanRequest scanRequest) {
         checkScanSubmitEmailDelivery(scanRequest);
 
         try {
-            Integer projectId;
+            Integer scanId;
             CxScanParams cxScanParams = scanRequestConverter.toScanParams(scanRequest);
-            Integer scanId = cxService.createScan(cxScanParams, "CxFlow Automated Scan");
-            projectId = cxScanParams.getProjectId();
+            Integer projectId = cxScanParams.getProjectId();
+
+            log.info("Checking if there is any existing scan for Project:" + projectId);
+            Integer existingScanId = cxService.getScanIdOfExistingScanIfExists(projectId);
+
+            if (existingScanId != UNKNOWN_INT) {
+                if (flowProperties.getScanResubmit()) {
+                    log.info("Existing ongoing scan with id " + existingScanId +" found for Project :" + projectId);
+                    log.info("Aborting the ongoing scan with id "+ existingScanId +" for Project:" + projectId);
+                    cxService.cancelScan(existingScanId);
+                    log.info("Resubmitting the scan for Project:" + projectId);
+                    scanId = cxService.createScan(cxScanParams, CXFLOW_SCAN_MSG);
+                } else {
+                      throw new CheckmarxException("Active Scan with Id " + existingScanId + " already exists for Project: " + projectId);
```

Comment:
Also String.format can be used here.
Probably add a message that this error is caused by a certain value of scan-resubmit? This will help user to understand where to look.

Reply:
Done!